enterprise/management/commands/update_enterprise_social_auth_uids.py

```diff
@@ -0,0 +1,165 @@
+        )
+        parser.add_argument(
+            '--new_prefix',
+            type=str,
+            default=None,
+            help='Optional new prefix for new UID. If not provided, uses CSV value.'
+        )
+        parser.add_argument(
+            '--no-dry-run',
+            action='store_false',
+            dest='dry_run',
+            default=True,
+            help='Actually save changes instead of simulating'
+        )
+
+    def handle(self, *args, **options):
+        logger.info("Command has started...")
+        csv_path = options['csv_file']
+        dry_run = options['dry_run']
+        old_prefix = options['old_prefix']
+        new_prefix = options['new_prefix']
+
+        total_processed = 0
+        total_updated = 0
+        total_errors = 0
+
+        try:
+            with open(csv_path, 'r') as csvfile:
+                reader = csv.DictReader(csvfile)
+
+                for row_num, row in enumerate(reader, start=1):
+                    total_processed += 1
+
+                    try:
+                        with transaction.atomic():
+                            if self.update_record(row, dry_run, old_prefix, new_prefix):
+                                total_updated += 1
+
+                    except Exception as row_error:  # pylint: disable=broad-except
+                        total_errors += 1
+                        error_msg = f"Row {row_num} update failed: {row} - Error: {str(row_error)}"
+                        logger.error(error_msg, exc_info=True)
+
+            summary_msg = (
+                f"CSV Update Summary:\n"
+                f"Total Records Processed: {total_processed}\n"
+                f"Records Successfully Updated: {total_updated}\n"
+                f"Errors Encountered: {total_errors}\n"
+                f"Dry Run Mode: {'Enabled' if dry_run else 'Disabled'}"
+            )
+            logger.info(summary_msg)
+        except IOError as io_error:
+            logger.critical(f"File I/O error: {str(io_error)}")
+
+        except Exception as e:  # pylint: disable=broad-except
+            logger.critical(f"Critical error in CSV processing: {str(e)}")
+
+    def update_record(self, row, dry_run=True, old_prefix=None, new_prefix=None):
+        """
+        Update a single record, applying optional prefixes to UIDs if provided.
+
+        Args:
+            row (dict): CSV row data
+            dry_run (bool): Whether to simulate or actually save changes
+            old_prefix (str): Prefix to apply to the old UID
+            new_prefix (str): Prefix to apply to the new UID
+
+        Returns:
+            bool: Whether the update was successful
+        """
+        try:
+            old_uid = row.get('old-uid')
+            new_uid = row.get('new-uid')
+
+            # Validating that both values are present
+            if not old_uid or not new_uid:
+                raise CSVUpdateError("Missing required UID fields")
```

Comment:
nit: should we also include required field names in raised exceptions to give a more meaningful message to user?

Reply:
After this raise, we are logging the complete row here which has the information about which field is missing.
`error_msg = f"Row {row_num} update failed: {row} - Error: {str(row_error)}"`